packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/executeQuery.ts

```diff
@@ -0,0 +1,415 @@
+                client,
+                complexType.set,
+                arrayValue,
+              )
+            ),
+          );
+
+          return new Set(remappedResponse);
+        }
+
+        case "struct": {
+          if (typeof responseValue !== "object") {
+            throw new Error(
+              `Expected object response, but received ${typeof responseValue}`,
+            );
+          }
+
+          const responseEntries = Object.entries(responseValue);
+          const remappedResponseEntries = await Promise.all(
+            responseEntries.map(async ([key, structValue]) => {
+              const structType = complexType.struct[key];
+              const remappedValue = await remapQueryResponseType(
+                client,
+                structType,
+                structValue,
+              );
+              return [key, remappedValue];
+            }),
+          );
+          const remappedResponse = remappedResponseEntries.reduce(
+            (acc, [key, mappedValue]) => {
+              acc[key as string] = mappedValue as ParameterValue;
+              return acc;
+            },
+            {} as { [key: string]: ParameterValue },
+          );
+
+          return remappedResponse;
+        }
+
+        case "twoDimensionalAggregation": {
+          const typedValue = responseValue as QueryTwoDimensionalAggregation;
+          const groups = typedValue.groups.map(group => {
+            const key = remapQueryBucketKeyType(
+              complexType.twoDimensionalAggregation,
+              group.key,
+            );
+            const value = remapQueryBucketValueType(
+              complexType.twoDimensionalAggregation.valueType,
+              group.value,
+            );
+            return {
+              key,
+              value,
+            };
+          });
+
+          return {
+            groups,
+          };
+        }
+
+        case "threeDimensionalAggregation": {
+          const typedValue = responseValue as QueryThreeDimensionalAggregation;
+          const groups = typedValue.groups.map(group => {
+            const key = remapQueryBucketKeyType(
+              complexType.threeDimensionalAggregation,
+              group.key,
+            );
+            const subBuckets = group.groups.map(subGroup => {
+              return {
+                key: remapQueryBucketKeyType(
+                  complexType.threeDimensionalAggregation.valueType,
+                  subGroup.key,
+                ),
+                value: remapQueryBucketValueType(
+                  complexType.threeDimensionalAggregation.valueType.valueType,
+                  subGroup.value,
+                ),
+              };
+            });
+            return {
+              key,
+              value: subBuckets,
+            };
+          });
+
+          return {
+            groups,
+          };
+        }
+
+        case "objectSet":
+          throw new Error("ObjectSet type not supported in response");
```

Comment:
We can handle this separately in a FLUP, but we can probably fill this in, just never got around to this in V0